### PR TITLE
Do not treat warnings as errors in solcjs

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
This should fix the pre-release issue and Solidity tests.

File: solcjs [hunk 1]
```diff
@@ -89,8 +89,17 @@ var output = solc.compile({ sources: sources }, argv.optimize ? 1 : 0);
 if (!output) {
   abort('No output from compiler');
 } else if (output['errors']) {
+  function isWarning (message) {
+    return message.match(/^(.*:[0-9]*:[0-9]* )?Warning: /)
```
Comment (axic) [inline solcjs]: This is the regexp from browser-solidity so it is "battle tested". Eventually this entire code should move onto jsonio alleviating this problem.

File: solcjs [hunk 2]
```diff
@@ -89,8 +89,17 @@ var output = solc.compile({ sources: sources }, argv.optimize ? 1 : 0);
...
+  }
+
   for (var error in output['errors']) {
-    console.error(output['errors'][error]);
+    var message = output['errors'][error]
+    if (isWarning(message)) {
+      console.log(message)
```
Comment (chriseth) [inline solcjs]: Does solcjs produce any other output on stdout? My feeling is that the data should go to stdout and anything user-facing through the "side-channel" stderr. But if we do not have output on stdout anyway (I didn't see anything), then this is fine.

Comment (axic) [reply solcjs]: Data goes into files.

**Comment by axic** [comment]
I'm merging this for the moment, but we should probably improve this repo a lot.

**Pull request merged**